src/stan/services/pathfinder/multi.hpp

```diff
@@ -105,126 +108,132 @@ inline int pathfinder_lbfgs_multi(
         tbb::blocked_range<int>(0, num_paths), [&](tbb::blocked_range<int> r) {
+          auto toss_write = [](auto&&... /* x */) {};
           for (int iter = r.begin(); iter < r.end(); ++iter) {
             auto pathfinder_ret
                 = stan::services::pathfinder::pathfinder_lbfgs_single<true>(
                     model, *(init[iter]), random_seed, stride_id + iter,
                     init_radius, history_size, init_alpha, tol_obj, tol_rel_obj,
                     tol_grad, tol_rel_grad, tol_param, num_iterations,
                     num_elbo_draws, num_draws, save_iterations, refresh,
-                    interrupt, logger, init_writers[iter],
-                    single_path_parameter_writer[iter],
-                    single_path_diagnostic_writer[iter], calculate_lp);
+                    interrupt, logger, init_writers[iter], toss_write,
+                    single_path_diagnostic_writer[iter], calculate_lp,
+                    psis_resample);
+            lp_calls += std::get<2>(pathfinder_ret);
             if (unlikely(std::get<0>(pathfinder_ret) != error_codes::OK)) {
               logger.error(std::string("Pathfinder iteration: ")
                            + std::to_string(iter) + " failed.");
               return;
             }
-            individual_lp_ratios[iter] = std::move(std::get<1>(pathfinder_ret));
-            individual_samples[iter] = std::move(std::get<2>(pathfinder_ret));
-            lp_calls += std::get<3>(pathfinder_ret);
+            elbo_estimates.push_back(
+                std::make_pair(iter, std::move(std::get<1>(pathfinder_ret))));
           }
         });
   } catch (const std::exception& e) {
     logger.error(e.what());
     return error_codes::SOFTWARE;
   }
-
-  // if any pathfinders failed, we want to remove their empty results
-  individual_lp_ratios.erase(
-      std::remove_if(individual_lp_ratios.begin(), individual_lp_ratios.end(),
-                     [](const auto& v) { return v.size() == 0; }),
-      individual_lp_ratios.end());
-  individual_samples.erase(
-      std::remove_if(individual_samples.begin(), individual_samples.end(),
-                     [](const auto& v) { return v.size() == 0; }),
-      individual_samples.end());
-
-  const auto end_pathfinders_time = std::chrono::steady_clock::now();
-
-  const double pathfinders_delta_time = stan::services::util::duration_diff(
-      start_pathfinders_time, end_pathfinders_time);
+  double pathfinders_delta_time = stan::services::util::duration_diff(
+      start_pathfinders_time, std::chrono::steady_clock::now());
   const auto start_psis_time = std::chrono::steady_clock::now();
-  const size_t successful_pathfinders = individual_samples.size();
-  if (successful_pathfinders == 0) {
-    logger.info("No pathfinders ran successfully");
-    return error_codes::SOFTWARE;
-  }
-  if (refresh != 0) {
-    logger.info("Total log probability function evaluations:"
-                + std::to_string(lp_calls));
```

Comment:
We're no longer printing this anywhere that I can see -- which might be fine, but in which case we can remove the `lp_calls` variable

Reply:
deleted